### PR TITLE
[feature] Remove added risk summary and response action options

### PR DESCRIPTION
**Pull request by deeonwuli**
### :pushpin: References

-   **Issue:** Closes https://app.clickup.com/t/8696j3edq

### :memo: Implementation
- [x] Remove added risk summary custom questions 
- [x] Remove added response actions
- [x] Show risk assessor selection

### :fire: Notes to the tester

#8696j3edq

### CODE REVIEW

File: src/data/repositories/IncidentActionD2Repository.ts [hunk 1]
```diff
@@ -178,13 +179,34 @@ export class IncidentActionD2Repository implements IncidentActionRepository {
...
             });
         });
     }
 
+    private deleteIncidentResponseAction(events: Id[]): FutureData<void> {
+        return apiToFuture(
+            this.api.tracker.post(
+                { importStrategy: "DELETE" },
+                // an array of event ids is sufficient for TrackerPostRequest with DELETE importStrategy
+                // @ts-ignore
```
Comment (anagperal) [inline src/data/repositories/IncidentActionD2Repository.ts:198]: (not need to change) In these cases another thing we usually do is to create the D2TrackerEvent like this:

```
        const d2Events: D2TrackerEvent[] = events.map(event => ({
            event: event,
            status: "COMPLETED",
            program: RTSL_ZEBRA_PROGRAM_ID,
            orgUnit: RTSL_ZEBRA_ORG_UNIT_ID,
            occurredAt: "",
            dataValues: [],
        }));
```

File: src/webapp/pages/form-page/useForm.ts [hunk 2]
```diff
@@ -104,6 +111,7 @@ export function useForm(formType: FormType, id?: Id): State {
                             isIncidentManager: isIncidentManager,
                         }),
                     });
+                    setEntityData(formData);
```
Comment (9sneha-n) [inline src/webapp/pages/form-page/useForm.ts:114]: isn't formdata already set in setConfigurableForm ? why not reuse it?

**Review by 9sneha-n — APPROVED** [review]
changes requested, not major enough to block merge/ Can be addressed in v2.

File: src/webapp/pages/form-page/useForm.ts [hunk 3]
```diff
@@ -226,6 +234,84 @@ export function useForm(formType: FormType, id?: Id): State {
...
+                    break;
+            }
+
+            setFormState(prevState => {
+                if (prevState.kind === "loaded") {
+                    const formSections = prevState.data.sections.filter(
+                        section => section.id !== id
+                    );
+
+                    return {
```
Comment (anagperal) [inline src/webapp/pages/form-page/useForm.ts:300]: Here we should check if form is valid with the new sections to disable or not the save button

[Screencast from 2024-11-28 12-03-12.webm](https://github.com/user-attachments/assets/10017013-6332-484f-b9c8-a3c5da3e23ad)

File: src/webapp/pages/form-page/useForm.ts (hunk 2 above)
Comment (deeonwuli) [reply src/webapp/pages/form-page/useForm.ts:114]: i'm updating `entityData` every time a form option is removed so i can get the `entityId` using the index. when i attempted to do this using `formData`, i wouldn't get the correct entityId and i would run into an error `onPrimaryButtonClick`

File: src/data/repositories/IncidentActionD2Repository.ts (hunk 1 above)
Comment (deeonwuli) [reply src/data/repositories/IncidentActionD2Repository.ts:198]: oh i see. noted!

**Review by anagperal — CHANGES_REQUESTED** [review]
thanks @deeonwuli !  Only one important comment: validate also the formState when something is deleted

**Review by anagperal — APPROVED** [review]
thanks @deeonwuli !!

**Review by bhavananarayanan — APPROVED** [review]
Approved !

**Pull request merged**